### PR TITLE
Panic equivalence

### PR DESCRIPTION
**Pull request by jakubadamw**
The continuation of #6.

First of all, thank you, @Soveu, for your changes, and @Plecra and @Shnatsel – for your input and past attempts. 🙂 

This is essentially a slightly corrected and somewhat cleaned-up version of #6. I also made it carry over the debug representations of compared values when `cfg(fuzzing_debug)` is enabled.

### CODE REVIEW

File: lib/src/lib.rs [hunk 1]
```diff
@@ -282,27 +279,110 @@ impl<'s> quote::ToTokens for MethodTest<'s> {
...
+                    struct GalaxyBrain<'a> {
+                        value: WhichFailed,
+                        to_update: &'a mut WhichFailed,
+                    }
+
+                    impl<'a> Drop for GalaxyBrain<'a> {
+                        fn drop(&mut self) {
+                            *self.to_update = self.value.clone();
+                        }
+                    }
```
Comment (Soveu) [inline lib/src/lib.rs:326]: To avoid unnecessary clones
```rust
std::mem::swap(self.to_update, &mut self.value);
```

**Comment by Shnatsel** [comment]
Dumb question, but: did anyone actually get this to work? I simply use `cargo hfuzz run`, it aborts the moment a panic occurs, and I don't understand why. There is no `panic=abort` on rustc command line when running `cargo hfuzz build`, and the fuzz target doesn't set `panic=abort` in Cargo.toml either.

To reproduce, comment out these lines which skip too-large inputs: 

https://github.com/Lokathor/tinyvec/blob/5758549693b979d9aff3105c062bc852062c0c26/fuzz/src/bin/tinyvec.rs#L50-L53

**Comment by jakubadamw** [comment]
@Shnatsel, oh dear. Yeah, this should have been tested. 🙂 

It seems like https://github.com/rust-fuzz/honggfuzz-rs/blob/master/src/lib.rs#L261-L271 is the culprit. It won't be hard to update the hook such that it does not abort on panics arising from within the inner closures.

On the other hand, honggfuzz-rs does not install the hook in the debug mode, where I am also seeing an abort, which is strange.

**Comment by jakubadamw** [comment]
@Shnatsel, indeed, adding a `std::panic::take_hook();` to the fuzz cycle solves the issue, though we really ought to install a hook that ignores the panics from within the closures. I'll try to fix this now. 🙂

**Comment by jakubadamw** [comment]
@Shnatsel, this is fixed now. Try updating to 0.2.1.

**Comment by Shnatsel** [comment]
It works now, thanks!

I've tested that this doesn't trip up when both implementations panic, but still registers as a crash if the tested implementation panics while the model does not.

**Pull request merged**